packages/screenreaderonly/react.d.ts

```diff
@@ -0,0 +1,24 @@
+/// <reference types="react" />
```

Comment:
Epic types.  I couldn't tell you if they're right or not.  Perhaps you could explain them.

Reply:
yep these are the gnarliest types i've written. there is a prop on the component call `as` that allows a string with the element name. these type enforce that the `as` prop is a valid html or svg element. also any additional html or react attributes have to match what is valid for the html or svg element.

in typescript this could potentially replace the filterReactProps. it's different but accomplishing a similar need